sasautos/h54s.sas

```diff
@@ -475,85 +475,47 @@ options NOQUOTELENMAX LRECL=32000 spool;
     %return;
   %end;
 
-
-  data _null_;
-    call symput('qc', '"');
-    call symput('pf', "%upcase(&dsn)");
-    call symput('dc', '$');
-    call symput('dt_', 'dt_');
+  proc contents noprint data=&libn..&dsn out=tempCols(keep=name type length);
```

Comment:
Use of proc contents instead of dictionary tables.  Faster, but two issues:
1 - the TYPE column is numeric (1 is num, 2 is char), which meant re-write further down
2 - the variables appear in a random order (not sorted on vartype)